# Projeto MapBiomas: Pipeline de Desmatamento Municipal

## MVP: Engenharia de Dados

**Autor:** Laura Cardoso Gonzaga  
**Matrícula:** 4052025001151  
**Fonte dos dados:** [MapBiomas Brasil – Desmatamento Municípios (Coleção 10)](https://brasil.mapbiomas.org/estatisticas/)

## Objetivo

O projeto visa analisar padrões de desmatamento no Brasil com base nos dados do **MapBiomas** utilizando um pipeline de engenharia de dados em Databricks. O MapBiomas é uma rede global e multi-institucional, formada por universidades, ONGs e empresas de tecnologia, que monitora as transformações na cobertura e no uso da terra nos territórios e seus impactos. 

O foco está em transformar dados brutos a partir de uma arquitetura medalhão (camadas bronze, prata e ouro) em uma modelagem estrela confiável para análise. A partir disso, o projeto busca responder às seguintes perguntas:

- Quais biomas concentram as maiores áreas desmatadas ao longo do tempo?

- Quais municípios apresentaram os maiores volumes de desmatamento acumulado?

- Como evoluiu o desmatamento no Brasil ao longo dos anos?

- Que tipos de vegetação foram mais afetados?

- Qual a proporção de supressão de vegetação primária em relação à secundária?


## Importação de bibliotecas necessárias

In [0]:
%pip install xlrd==2.0.1 openpyxl

import pandas as pd
from pyspark.sql.functions import col

Note: you may need to restart the kernel using %restart_python or dbutils.library.restartPython() to use updated packages.


## Estrutura do Catálogo `mapbiomas`

In [0]:
'''
mapbiomas/
├── default/                 
│   └── data/                  ← Volume com dados originais
│       ├── mapbiomas_deforestation.xlsx
│       └── class_legend.csv
│       └── AR_BR_RG_UF_RGINT_RGI_MUN_2024.xls
│
├── bronze/                   ← Dados crus em formato Delta
│   └── mapbiomas_deforestation
│
├── silver/                   ← Dados transformados (stack, joins, cast, etc)
│   └── mapbiomas_deforestation
│
├── gold/                     ← Tabelas fato e dimensão
│   ├── fact_desmatamento
│   ├── dim_municipio
│   └── dim_classe_vegetacao
│
└── dim/                      ← Tabelas extras (legenda e dados externos)
    └── class_legend
    └── area_estados
'''

'\nmapbiomas/\n├── default/                 \n│   └── data/                  ← Volume com dados originais\n│       ├── mapbiomas_deforestation.xlsx\n│       └── class_legend.csv\n│       └── AR_BR_RG_UF_RGINT_RGI_MUN_2024.xls\n│\n├── bronze/                   ← Dados crus em formato Delta\n│   └── mapbiomas_deforestation\n│\n├── silver/                   ← Dados transformados (stack, joins, cast, etc)\n│   └── mapbiomas_deforestation\n│\n├── gold/                     ← Tabelas fato e dimensão\n│   ├── fact_desmatamento\n│   ├── dim_municipio\n│   └── dim_classe_vegetacao\n│\n└── dim/                      ← Tabelas extras (legenda e dados externos)\n    └── class_legend\n    └── area_estados\n'

In [0]:
%sql
USE CATALOG mapbiomas;

DROP SCHEMA IF EXISTS bronze CASCADE;
CREATE SCHEMA bronze;
DROP SCHEMA IF EXISTS silver CASCADE;
CREATE SCHEMA silver;
DROP SCHEMA IF EXISTS gold CASCADE;
CREATE SCHEMA gold;
DROP SCHEMA IF EXISTS dim CASCADE;
CREATE SCHEMA dim;

## Ingestão (Camada Bronze)

Nesta etapa, criamos a camada bronze com os arquivos raw, preservando os dados originais sem transformação. O objetivo é garantir reprodutibilidade e rastreabilidade, mantendo os dados brutos acessíveis e separados das demais etapas do pipeline.

#### Leitura dos dados brutos e escrita dos dados como tabelas Delta

In [0]:
df = pd.read_excel("/Volumes/mapbiomas/default/data/MAPBIOMAS_BRAZIL_DEFORESTATION_MUNICIPALITIES_COL.10_v2.xlsx",
                   sheet_name="DEFORESTATION") # tabela com os dados estruturados

df.head()

country     biome state  ...         2022         2023         2024
0  Brasil  Amazônia  Acre  ...  2708.697398  2049.721612  1304.135691
1  Brasil  Amazônia  Acre  ...   259.623443   367.193797    21.231191
2  Brasil  Amazônia  Acre  ...     8.901130     3.964020     1.673282
3  Brasil  Amazônia  Acre  ...     0.528165     6.252777     0.440155
4  Brasil  Amazônia  Acre  ...     0.000000     0.176145     0.000000

[5 rows x 50 columns]

In [0]:
df.isna().any()

country                 False
biome                   False
state                   False
municipality            False
geocode_municipality    False
class                   False
transition_name         False
class_level_0           False
class_level_1           False
class_level_2           False
class_level_3           False
class_level_4           False
1987                    False
1988                    False
1989                    False
1990                    False
1991                    False
1992                    False
1993                    False
1994                    False
1995                    False
1996                    False
1997                    False
1998                    False
1999                    False
2000                    False
2001                    False
2002                    False
2003                    False
2004                    False
2005                    False
2006                    False
2007                    False
2008      

In [0]:
df_bronze = spark.createDataFrame(df)

df_bronze.write.format("delta").mode("overwrite").saveAsTable("bronze.mapbiomas_deforestation")

In [0]:
df_bronze.printSchema()

root
 |-- country: string (nullable = true)
 |-- biome: string (nullable = true)
 |-- state: string (nullable = true)
 |-- municipality: string (nullable = true)
 |-- geocode_municipality: long (nullable = true)
 |-- class: long (nullable = true)
 |-- transition_name: string (nullable = true)
 |-- class_level_0: string (nullable = true)
 |-- class_level_1: string (nullable = true)
 |-- class_level_2: string (nullable = true)
 |-- class_level_3: string (nullable = true)
 |-- class_level_4: string (nullable = true)
 |-- 1987: double (nullable = true)
 |-- 1988: double (nullable = true)
 |-- 1989: double (nullable = true)
 |-- 1990: double (nullable = true)
 |-- 1991: double (nullable = true)
 |-- 1992: double (nullable = true)
 |-- 1993: double (nullable = true)
 |-- 1994: double (nullable = true)
 |-- 1995: double (nullable = true)
 |-- 1996: double (nullable = true)
 |-- 1997: double (nullable = true)
 |-- 1998: double (nullable = true)
 |-- 1999: double (nullable = true)
 |-- 2000: do

#### Legenda com tradução para o português

In [0]:
class_legend = pd.read_csv("/Volumes/mapbiomas/default/data/Codigos-da-legenda-colecao-10.csv",
                           delimiter= "\t")

class_legend.head(20)

Class_ID  Level  ...                       Descricao    Color
0          1      1  ...                        Floresta  #32a65e
1          3      2  ...              Formação Florestal  #1f8d49
2          4      2  ...               Formação Savânica  #7dc975
3          5      2  ...                          Mangue  #04381d
4          6      2  ...               Floresta Alagável  #026975
5         49      2  ...                Restinga Arbórea  #02d659
6         10      1  ...  Vegetação Herbácea e Arbustiva  #ad975a
7         11      2  ...  Campo Alagado e Área Pantanosa  #519799
8         12      2  ...              Formação Campestre  #d6bc74
9         32      2  ...                          Apicum  #fc8114
10        29      2  ...             Afloramento Rochoso  #ffaa5f
11        50      2  ...               Restinga Herbácea  #ad5100
12        14      1  ...                    Agropecuária  #FFFFB2
13        15      2  ...                        Pastagem  #edde8e
14        18      2  ...                     Agricultura  #E974ED
15        19      3  ...              Lavoura Temporária  #C27BA0
16        39      4  ...                            Soja  #f5b3c8
17        20      4  ...                            Cana  #db7093
18        40      4  ...                           Arroz  #c71585
19        62      4  ...                  Algodão (beta)  #ff69b4

[20 rows x 5 columns]

In [0]:
df_legenda_spark = spark.createDataFrame(class_legend)

df_legenda_spark.write.format("delta").mode("overwrite").saveAsTable("dim.class_legend")

#### Dados externos IBGE: área dos estados

As áreas dos estados foram obtidas do [IBGE (2024)](https://www.ibge.gov.br/geociencias/organizacao-do-territorio/estrutura-territorial/15761-areas-dos-municipios.html) e convertidas de km² para hectares.

In [0]:
df_area_estados = pd.read_excel(
    "/Volumes/mapbiomas/default/data/AR_BR_RG_UF_RGINT_RGI_MUN_2024.xls",
    sheet_name="AR_BR_UF_2024"
)

In [0]:
area_estados_spark = spark.createDataFrame(df_area_estados)

In [0]:
area_estados_min = (
    area_estados_spark
    .selectExpr(
        "NM_UF AS state",
        "CAST(AR_UF_2024 * 100 AS DOUBLE) AS area_uf_ha"     #  1 km² = 100 ha
    )
    .dropDuplicates(["state"])
    .filter(col("state").isNotNull())
)

area_estados_min.write.mode("overwrite").saveAsTable("dim.area_estados")

#### Metadados

In [0]:
metadados = pd.read_excel(
    "/Volumes/mapbiomas/default/data/MAPBIOMAS_BRAZIL_DEFORESTATION_MUNICIPALITIES_COL.10_v2.xlsx",
    sheet_name="METADADOS",
    header=2,
    index_col=0,
    usecols=range(1,4)
)

metadados.head(20)

Descrição do campo                                Field Description
Campo/Field                                                                                                             
country                                                            País                                          Country
biome                                                             Bioma                                            Biome
state                                                            Estado                                            State
municipality                                                  Município                                     Municipality
geocode_municipality                           Código do município IBGE                         Municipality code (IBGE)
class                 Código da legenda correspondente a classe do m...  Legend code for each deforestation module class
transition_name                Nome da classe do módulo de desmatamento           Name of the deforestation module class
class_level_1             Classe de cobertura e uso da terra no nível 1  Land cover and land use class on legend level 1
class_level_2             Classe de cobertura e uso da terra no nível 2  Land cover and land use class on legend level 2
class_level_3             Classe de cobertura e uso da terra no nível 3  Land cover and land use class on legend level 3
class_level_4             Classe de cobertura e uso da terra no nível 4  Land cover and land use class on legend level 4

## Transformação (Camada Silver)

A camada silver aplica as primeiras transformações estruturais como limpeza, padronização e enriquecimento com metadados. Aqui já obtemos uma tabela utilizável para análise, mas ainda não modelada para consulta analítica rápida.


In [0]:
%sql
DESCRIBE bronze.mapbiomas_deforestation;

col_name data_type comment country string null biome string null state string null municipality string null geocode_municipality bigint null class bigint null transition_name string null class_level_0 string null class_level_1 string null class_level_2 string null class_level_3 string null class_level_4 string null 1987 double null 1988 double null 1989 double null 1990 double null 1991 double null 1992 double null 1993 double null 1994 double null 1995 double null 1996 double null 1997 double null 1998 double null 1999 double null 2000 double null 2001 double null 2002 double null 2003 double null 2004 double null 2005 double null 2006 double null 2007 double null 2008 double null 2009 double null 2010 double null 2011 double null 2012 double null 2013 double null 2014 double null 2015 double null 2016 double null 2017 double null 2018 double null 2019 double null 2020 double null 2021 double null 2022 double null 2023 double null 2024 double null

#### Conversão de colunas wide para long com `stack()`
Identificação automática das colunas de ano para transformar em um única coluna 'year'. Dessa forma, cada linha corresponde a uma observação, facilitando a agregação para análise.

In [0]:
year_cols = [c for c in df_bronze.columns if c.isdigit() and len(c) == 4]
year_cols = sorted(year_cols)

print(year_cols)

['1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023', '2024']


In [0]:
stack_expr = "stack({n}, {pairs}) as (year, area_ha)".format(
    n=len(year_cols),
    pairs=", ".join([f"'{y}', `{y}`" for y in year_cols])
)

#### Observando os dados
Agregação de categorias de classe conclui que 'class_level_2', 'class_level_3' e 'class_level_4' são reduntantes. Portanto, podemos manter apenas uma.

In [0]:
df_bronze.select("class", "class_level_0", "class_level_1", "class_level_2", "class_level_3", "class_level_4") \
    .distinct() \
    .orderBy("class") \
    .display()

class class_level_0 class_level_1 class_level_2 class_level_3 class_level_4 3 Natural 1. Forest 1.1. Forest Formation 1.1. Forest Formation 1.1. Forest Formation 4 Natural 1. Forest 1.2. Savanna Formation 1.2. Savanna Formation 1.2. Savanna Formation 5 Natural 1. Forest 1.3. Mangrove 1.3. Mangrove 1.3. Mangrove 6 Natural 1. Forest 1.4 Floodable Forest 1.4 Floodable Forest 1.4 Floodable Forest 11 Natural 2. Non Forest Natural Formation 2.1. Wetland 2.1. Wetland 2.1. Wetland 12 Natural 2. Non Forest Natural Formation 2.2. Grassland 2.2. Grassland 2.2. Grassland 13 Natural 2. Non Forest Natural Formation 2.6. Other non Forest Formations 2.6. Other non Forest Formations 2.6. Other non Forest Formations 49 Natural 1. Forest 1.5. Wooded Sandbank Vegetation 1.5. Wooded Sandbank Vegetation 1.5. Wooded Sandbank Vegetation 50 Natural 2. Non Forest Natural Formation 2.4. Herbaceous Sandbank Vegetation 2.4. Herbaceous Sandbank Vegetation 2.4. Herbaceous Sandbank Vegetation

Brasil é o único país presente nos dados, logo essa também pode ser uma coluna descartável.

In [0]:
df_bronze.select("country").distinct().display()

country Brasil

#### Criação de dataframe e tabela

In [0]:
dim_cols = [
    "geocode_municipality", # PK 
    "biome",
    "state",
    "municipality",
    "class",
    "transition_name",
    "class_level_1",
    "class_level_2"
]

In [0]:
df_silver = df_bronze.select(*dim_cols, *year_cols) \
    .selectExpr(*dim_cols, stack_expr) \
    .withColumn("year", col("year").cast("int"))

display(df_silver.limit(10))

geocode_municipality biome state municipality class transition_name class_level_1 class_level_2 year area_ha 1200013 Amazônia Acre Acrelândia 3 Supressão Veg. Primária 1. Forest 1.1. Forest Formation 1987 3069.302624029574 1200013 Amazônia Acre Acrelândia 3 Supressão Veg. Secundária 1. Forest 1.1. Forest Formation 1987 0.0 1200013 Amazônia Acre Acrelândia 6 Supressão Veg. Primária 1. Forest 1.4 Floodable Forest 1987 2.81810064086914 1200013 Amazônia Acre Acrelândia 6 Supressão Veg. Secundária 1. Forest 1.4 Floodable Forest 1987 0.0 1200013 Amazônia Acre Acrelândia 11 Supressão Veg. Primária 2. Non Forest Natural Formation 2.1. Wetland 1987 7.575158142089847 1200013 Amazônia Acre Acrelândia 11 Supressão Veg. Secundária 2. Non Forest Natural Formation 2.1. Wetland 1987 0.0 1200013 Amazônia Acre Acrelândia 12 Supressão Veg. Primária 2. Non Forest Natural Formation 2.2. Grassland 1987 10.83564494018555 1200013 Amazônia Acre Acrelândia 12 Supressão Veg. Secundária 2. Non Forest Natural Formation 2.2. Grassland 1987 0.0 1200054 Amazônia Acre Assis Brasil 3 Supressão Veg. Primária 1. Forest 1.1. Forest Formation 1987 221.3526631408689 1200054 Amazônia Acre Assis Brasil 3 Supressão Veg. Secundária 1. Forest 1.1. Forest Formation 1987 0.0

In [0]:
df_silver.write.format("delta").mode("overwrite") \
    .option("mergeSchema", "true") \
    .saveAsTable("silver.mapbiomas_deforestation")
        

In [0]:
%sql
SELECT COUNT(*) as total,
       SUM(CASE WHEN area_ha IS NULL THEN 1 ELSE 0 END) AS nulos,
       SUM(CASE WHEN area_ha < 0 THEN 1 ELSE 0 END) AS negativos
FROM silver.mapbiomas_deforestation

total nulos negativos 1401782 0 0

#### Inclusão de nomes em português:
 Criação de colunas de classe florestal em português `class_level_1_pt` e `class_level_2_pt` com base em `class_legend`. A legenda não incluía a categoria "Outras Formações não Florestais" e foi incluída manualmente com CASE WHEN.



In [0]:
%sql
CREATE OR REPLACE TABLE silver.mapbiomas_deforestation AS
SELECT s.biome, s.state, s.municipality, s.geocode_municipality, s.transition_name, 
       s.class_level_1, 
       CASE WHEN s.class_level_1 = '1. Forest' THEN 'Floresta'
            WHEN s.class_level_1 = '2. Non Forest Natural Formation' THEN 'Formação não Florestal'
            ELSE NULL
            END AS class_level_1_pt,
       s.class_level_2, 
       CASE WHEN l.Descricao IS NOT NULL THEN l.Descricao
            WHEN s.class_level_2 = '2.6. Other non Forest Formations' THEN 'Outras Formações não Florestais'
            ELSE NULL
            END AS class_level_2_pt, 
       s.year, s.area_ha
FROM silver.mapbiomas_deforestation as s
LEFT JOIN dim.class_legend as l
ON s.class = l.Class_ID

num_affected_rows num_inserted_rows

In [0]:
df_silver = spark.read.table("silver.mapbiomas_deforestation")

df_silver.groupBy("class_level_1", "class_level_1_pt", "class_level_2", "class_level_2_pt") \
    .count() \
    .orderBy("class_level_1") \
    .display()

class_level_1 class_level_1_pt class_level_2 class_level_2_pt count 1. Forest Floresta 1.1. Forest Formation Formação Florestal 471656 1. Forest Floresta 1.3. Mangrove Mangue 18316 1. Forest Floresta 1.5. Wooded Sandbank Vegetation Restinga Arbórea 17594 1. Forest Floresta 1.2. Savanna Formation Formação Savânica 288572 1. Forest Floresta 1.4 Floodable Forest Floresta Alagável 42712 2. Non Forest Natural Formation Formação não Florestal 2.2. Grassland Formação Campestre 271662 2. Non Forest Natural Formation Formação não Florestal 2.1. Wetland Campo Alagado e Área Pantanosa 275234 2. Non Forest Natural Formation Formação não Florestal 2.4. Herbaceous Sandbank Vegetation Restinga Herbácea 15960 2. Non Forest Natural Formation Formação não Florestal 2.6. Other non Forest Formations Outras Formações não Florestais 76

#### Adicionando metadados como comentários 

In [0]:
def set_comentario(table_name, metadados):
    table_cols = {row['col_name'] for row in spark.sql(f"DESCRIBE {table_name}").collect()}
    # Colunas não presentes no dataframe de metadados
    extras = {
        "class_level_1_pt": metadados.at["class_level_1", metadados.columns[0]] + " (em português)",
        "class_level_2_pt": metadados.at["class_level_2", metadados.columns[0]] + " (em português)",
        "year": "Ano",
        "area_ha": "Área em hectares"
    }
    comentarios = {campo: comentario for campo, comentario in zip(metadados.index, metadados.iloc[:,0])}
    comentarios.update(extras)
    for campo, comentario in comentarios.items():
        if campo in table_cols:
            spark.sql(f"ALTER TABLE {table_name} ALTER COLUMN `{campo}` COMMENT '{comentario}'")

set_comentario("silver.mapbiomas_deforestation", metadados)

In [0]:
%sql
DESCRIBE silver.mapbiomas_deforestation;

col_name data_type comment biome string Bioma state string Estado municipality string Município geocode_municipality bigint Código do município IBGE transition_name string Nome da classe do módulo de desmatamento class_level_1 string Classe de cobertura e uso da terra no nível 1 class_level_1_pt string Classe de cobertura e uso da terra no nível 1 (em português) class_level_2 string Classe de cobertura e uso da terra no nível 2 class_level_2_pt string Classe de cobertura e uso da terra no nível 2 (em português) year int Ano area_ha double Área em hectares

## Modelagem Estrela (Camada Gold)

Na camada gold, reestruturamos os dados para análises otimizadas usando o modelo estrela (star schema), separando tabelas fato (valores quantitativos) e dimensão (atributos descritivos). A tabela fato guarda chaves estrangeiras que apontam para as dimensões. Essa estrutura facilita agregações, cruzamentos e visualizações com desempenho e clareza.

**Padrão de chaves**
- geocode_municipality → código IBGE
- state → nome completo (ex.: “Mato Grosso”)
- biome → nome conforme MapBiomas

![Modelo ER](https://github.com/lauragonzaga/MVP-Engenharia-de-Dados/blob/main/diagrama_ER.png?raw=true)

#### Tabelas Dimensão

In [0]:
%sql
CREATE OR REPLACE TABLE gold.dim_municipio AS
SELECT DISTINCT 
  geocode_municipality,
  municipality,
  state,
  biome
FROM silver.mapbiomas_deforestation

num_affected_rows num_inserted_rows

In [0]:
%sql
CREATE OR REPLACE TABLE gold.dim_classe_vegetacao AS
SELECT DISTINCT
  hash(class_level_1, class_level_2) AS classe_id,
  class_level_1_pt,
  class_level_2_pt
FROM silver.mapbiomas_deforestation;

num_affected_rows num_inserted_rows

In [0]:
%sql
CREATE OR REPLACE TABLE gold.dim_transicao AS
SELECT DISTINCT
  hash(transition_name) AS transicao_id,
  transition_name
FROM silver.mapbiomas_deforestation
WHERE transition_name IS NOT NULL;

num_affected_rows num_inserted_rows

#### Tabela Fato

In [0]:
%sql
CREATE OR REPLACE TABLE gold.fact_desmatamento AS
SELECT 
  geocode_municipality,
  hash(class_level_1, class_level_2) AS classe_id,
  hash(transition_name) AS transicao_id,
  year,
  area_ha
FROM silver.mapbiomas_deforestation;

num_affected_rows num_inserted_rows

#### Checagem
Garantindo integridade conferindo se há chaves órfãs.

In [0]:
%sql
SELECT
  (SELECT COUNT(*) 
  FROM gold.fact_desmatamento f 
  LEFT ANTI JOIN gold.dim_municipio d 
  ON f.geocode_municipality = d.geocode_municipality) AS missing_municipios,
  (SELECT COUNT(*) 
  FROM gold.fact_desmatamento f 
  LEFT ANTI JOIN gold.dim_transicao d 
  ON f.transicao_id = d.transicao_id) AS missing_transicao,
  (SELECT COUNT(*) 
  FROM gold.fact_desmatamento f 
  LEFT ANTI JOIN gold.dim_classe_vegetacao d 
  ON f.classe_id = d.classe_id) AS missing_classes

missing_municipios missing_transicao missing_classes 0 0 0

## Exploração

Nesta etapa, utilizamos as tabelas da camada _gold_ para produzir consultas SQL que respondam às perguntas de negócio definidas no início. Essas consultas foram utitilizadas para construir um [dashboard](https://dbc-b08da042-4e74.cloud.databricks.com/dashboardsv3/01f0d08344e91b4791a28f4d268be1dd/published?o=1041144495999733) como produto final da análise.


In [0]:
%sql
-- Cobertura temporal
SELECT MIN(year) AS ano_inicial, MAX(year) AS ano_final
FROM gold.fact_desmatamento;

ano_inicial ano_final 1987 2024

In [0]:
%sql
-- Total de área nativa e acumulada desmatada no Brasil
SELECT ROUND(SUM(CASE WHEN transition_name = 'Supressão Veg. Primária' 
                 THEN area_ha ELSE 0 END),2) as area_nativa_desmatada,
      ROUND(SUM(area_ha),2) as area_desmatada_acumulada
FROM gold.fact_desmatamento g
JOIN gold.dim_transicao m
  ON g.transicao_id = m.transicao_id


area_nativa_desmatada area_desmatada_acumulada 1.4486848425E8 1.7622854824E8

In [0]:
%sql
-- Série temporal histórica nacional 

SELECT year, ROUND(SUM(area_ha), 2) AS area_total_ha
FROM gold.fact_desmatamento
GROUP BY year
ORDER BY year;

year area_total_ha 1987 5647033.74 1988 5580900.85 1989 5158285.02 1990 4848121.02 1991 4884244.17 1992 5760493.03 1993 5222493.16 1994 5459509.55 1995 6224311.54 1996 5611487.63 1997 5396760.43 1998 5485654.48 1999 5416146.56 2000 4979128.09 2001 4908308.28 2002 5096811.47 2003 6382510.13 2004 6267097.78 2005 5140749.99 2006 4354559.1 2007 3791482.92 2008 3900777.19 2009 3103783.9 2010 2994038.57 2011 3085666.45 2012 3509577.4 2013 3904947.23 2014 3449484.36 2015 3640508.66 2016 3394053.93 2017 3140599.28 2018 3211782.25 2019 4234172.99 2020 4246855.79 2021 4565834.37 2022 5012140.54 2023 5035415.36 2024 4182821.03

In [0]:
%sql
-- Evolução do desmatamento acumulado por bioma

SELECT biome, year, ROUND(SUM(area_ha),2) AS area_total_ha
FROM gold.fact_desmatamento g
JOIN gold.dim_municipio m
ON g.geocode_municipality = m.geocode_municipality
GROUP BY year, biome
ORDER BY year, biome

biome year area_total_ha Amazônia 1987 1975012.69 Caatinga 1987 696272.78 Cerrado 1987 2999296.45 Mata Atlântica 1987 1150831.0 Pampa 1987 209591.64 Pantanal 1987 193954.84 Amazônia 1988 1759695.24 Caatinga 1988 740738.18 Cerrado 1988 2856497.19 Mata Atlântica 1988 1203512.59 Pampa 1988 405717.77 Pantanal 1988 160532.54 Amazônia 1989 1773628.09 Caatinga 1989 559150.71 Cerrado 1989 2759808.11 Mata Atlântica 1989 1003679.11 Pampa 1989 305943.88 Pantanal 1989 188314.43 Amazônia 1990 1563461.04 Caatinga 1990 667040.06 Cerrado 1990 2608280.53 Mata Atlântica 1990 898960.16 Pampa 1990 250641.87 Pantanal 1990 183663.8 Amazônia 1991 1634811.17 Caatinga 1991 708175.77 Cerrado 1991 2464989.66 Mata Atlântica 1991 930204.85 Pampa 1991 276964.85 Pantanal 1991 163893.94 Amazônia 1992 2195362.18 Caatinga 1992 1006781.58 Cerrado 1992 2835252.75 Mata Atlântica 1992 835771.45 Pampa 1992 220316.23 Pantanal 1992 213467.0 Amazônia 1993 2094152.61 Caatinga 1993 895821.87 Cerrado 1993 2495783.57 Mata Atlântica 1993 769609.52 Pampa 1993 287946.3 Pantanal 1993 165935.43 Amazônia 1994 2446908.97 Caatinga 1994 756239.44 Cerrado 1994 2739541.13 Mata Atlântica 1994 773872.35 Pampa 1994 270417.75 Pantanal 1994 201724.88 Amazônia 1995 3137023.09 Caatinga 1995 805867.2 Cerrado 1995 2814516.68 Mata Atlântica 1995 748963.96 Pampa 1995 261777.18 Pantanal 1995 224046.5 Amazônia 1996 2881525.16 Caatinga 1996 724728.95 Cerrado 1996 2514329.48 Mata Atlântica 1996 696802.93 Pampa 1996 249634.13 Pantanal 1996 198957.27 Amazônia 1997 2829115.19 Caatinga 1997 698981.52 Cerrado 1997 2316238.07 Mata Atlântica 1997 668287.13 Pampa 1997 272593.08 Pantanal 1997 157952.19 Amazônia 1998 2830025.89 Caatinga 1998 823690.79 Cerrado 1998 2323687.8 Mata Atlântica 1998 631579.55 Pampa 1998 232952.7 Pantanal 1998 150701.38 Amazônia 1999 2624442.62 Caatinga 1999 823323.5 Cerrado 1999 2413909.11 Mata Atlântica 1999 688732.4 Pampa 1999 240567.46 Pantanal 1999 164528.33 Amazônia 2000 2327598.67 Caatinga 2000 752821.0 Cerrado 2000 2269885.85 Mata Atlântica 2000 601527.2 Pampa 2000 219157.41 Pantanal 2000 125999.86 Amazônia 2001 2527290.81 Caatinga 2001 739928.17 Cerrado 2001 2070052.19 Mata Atlântica 2001 522283.84 Pampa 2001 239147.43 Pantanal 2001 131024.84 Amazônia 2002 2741501.21 Caatinga 2002 746037.47 Cerrado 2002 2103184.03 Mata Atlântica 2002 535787.8 Pampa 2002 238685.67 Pantanal 2002 130386.87 Amazônia 2003 3772613.7 Caatinga 2003 711909.11 Cerrado 2003 2747721.81 Mata Atlântica 2003 604197.17 Pampa 2003 313810.67 Pantanal 2003 182928.92 Amazônia 2004 3531035.88 Caatinga 2004 810691.98 Cerrado 2004 2664447.05 Mata Atlântica 2004 585515.1 Pampa 2004 326042.87 Pantanal 2004 129070.8 Amazônia 2005 2901692.89 Caatinga 2005 708864.28 Cerrado 2005 2067546.8 Mata Atlântica 2005 488810.93 Pampa 2005 227999.99 Pantanal 2005 115499.8 Amazônia 2006 2319170.66 Caatinga 2006 693843.74 Cerrado 2006 1667111.35 Mata Atlântica 2006 461707.17 Pampa 2006 204597.11 Pantanal 2006 99457.57 Amazônia 2007 1896603.55 Caatinga 2007 648066.27 Cerrado 2007 1458406.05 Mata Atlântica 2007 409633.59 Pampa 2007 196964.04 Pantanal 2007 83649.74 Amazônia 2008 1823875.03 Caatinga 2008 687668.07 Cerrado 2008 1594927.77 Mata Atlântica 2008 439711.2 Pampa 2008 235998.62 Pantanal 2008 84903.6 Amazônia 2009 1184231.0 Caatinga 2009 662717.64 Cerrado 2009 1319558.91 Mata Atlântica 2009 426338.87 Pampa 2009 185113.06 Pantanal 2009 78177.88 Amazônia 2010 1074720.14 Caatinga 2010 611568.84 Cerrado 2010 1272002.05 Mata Atlântica 2010 427664.95 Pampa 2010 251201.2 Pantanal 2010 72629.19 Amazônia 2011 1128990.95 Caatinga 2011 672845.69 Cerrado 2011 1329819.84 Mata Atlântica 2011 411511.4 Pampa 2011 253434.68 Pantanal 2011 69695.24 Amazônia 2012 1125996.8 Caatinga 2012 775248.17 Cerrado 2012 1741475.15 Mata Atlântica 2012 425274.75 Pampa 2012 255262.96 Pantanal 2012 70463.45 Amazônia 2013 1353905.9 Caatinga 2013 785977.63 Cerrado 2013 1934892.45 Mata Atlântica 2013 468397.78 Pampa 2013 280304.1 Pantanal 2013 71182.85 Amazônia 

In [0]:
%sql
-- Top 10 estados com maior desmatamento acumulado
SELECT 
  m.state, 
  ROUND(SUM(g.area_ha),2) AS area_desmatada_acumulada
FROM gold.fact_desmatamento g
JOIN gold.dim_municipio m ON g.geocode_municipality = m.geocode_municipality
JOIN gold.dim_transicao t ON g.transicao_id = t.transicao_id
GROUP BY m.state
ORDER BY area_desmatada_acumulada DESC
LIMIT 10

state area_desmatada_acumulada Mato Grosso 4.523567371E7 Pará 2.970735232E7 Bahia 2.134230516E7 Minas Gerais 1.587357796E7 Maranhão 1.454250159E7 Rio Grande do Sul 1.302828602E7 Tocantins 1.299893727E7 Mato Grosso do Sul 1.184878251E7 Rondônia 1.03206583E7 Goiás 9502920.3

In [0]:
%sql
-- Top 10 estados com maior percentual de área nativa desmatada

WITH area_nativa_uf AS (
  SELECT 
    m.state,
    ROUND(SUM(CASE 
      WHEN t.transition_name = 'Supressão Veg. Primária' 
      THEN g.area_ha ELSE 0 END), 2) AS area_nativa_desmatada
  FROM gold.fact_desmatamento g
  JOIN gold.dim_transicao t 
    ON g.transicao_id = t.transicao_id
  JOIN gold.dim_municipio m 
    ON g.geocode_municipality = m.geocode_municipality
  GROUP BY m.state
)

SELECT 
  a.state,
  a.area_nativa_desmatada,
  ROUND(a.area_nativa_desmatada / e.area_uf_ha * 100, 2) AS perc_desmatado
FROM area_nativa_uf a
JOIN dim.area_estados e 
  ON a.state = e.state
ORDER BY perc_desmatado DESC
LIMIT 10


state area_nativa_desmatada perc_desmatado Mato Grosso 4.055999277E7 44.91 Rondônia 9665801.63 40.65 Tocantins 1.072780893E7 38.67 Maranhão 1.185536503E7 35.96 Rio Grande do Sul 8667168.32 30.77 Bahia 1.665009897E7 29.48 Mato Grosso do Sul 1.040619055E7 29.14 Sergipe 588830.72 26.84 Ceará 3860336.07 25.93 Piauí 6422651.4 25.51

In [0]:
%sql
-- Top 10 municípios com maior desmatamento acumulado

SELECT CONCAT(municipality, ', ', state) AS municipio, 
       ROUND(SUM(area_ha), 2) AS area_total_ha
FROM gold.fact_desmatamento g
JOIN gold.dim_municipio m
ON g.geocode_municipality = m.geocode_municipality
GROUP BY m.municipality, m.state
ORDER BY area_total_ha DESC
LIMIT 10

municipio area_total_ha São Félix do Xingu, Pará 2138865.15 Paranatinga, Mato Grosso 1802867.29 Cáceres, Mato Grosso 1774184.43 Santa Maria das Barreiras, Pará 1311207.8 Santana do Araguaia, Pará 1296521.18 Brasnorte, Mato Grosso 1229349.12 Porto Velho, Rondônia 1224879.52 Altamira, Pará 1219260.1 Nova Ubiratã, Mato Grosso 1162296.0 Corumbá, Mato Grosso do Sul 1066775.57

In [0]:
%sql
-- Tipo de vegetação mais desmatado

SELECT class_level_1_pt, c.class_level_2_pt, ROUND(SUM(f.area_ha), 2) AS area_total_ha
FROM gold.fact_desmatamento f
JOIN gold.dim_classe_vegetacao c ON f.classe_id = c.classe_id
GROUP BY class_level_1_pt, c.class_level_2_pt
ORDER BY area_total_ha DESC;

class_level_1_pt class_level_2_pt area_total_ha Floresta Formação Florestal 8.684739797E7 Floresta Formação Savânica 6.881205983E7 Formação não Florestal Formação Campestre 1.334650823E7 Floresta Floresta Alagável 3640076.95 Formação não Florestal Campo Alagado e Área Pantanosa 3149397.26 Floresta Restinga Arbórea 229834.8 Formação não Florestal Restinga Herbácea 198709.06 Floresta Mangue 4562.47 Formação não Florestal Outras Formações não Florestais 1.68

In [0]:
%sql
-- Supressão primária vs. secundária

SELECT t.transition_name, ROUND(SUM(f.area_ha), 2) AS area_total_ha
FROM gold.fact_desmatamento f
JOIN gold.dim_transicao t ON f.transicao_id = t.transicao_id
GROUP BY t.transition_name
ORDER BY area_total_ha DESC;

transition_name area_total_ha Supressão Veg. Primária 1.4486848425E8 Supressão Veg. Secundária 3.136006399E7

In [0]:
%sql
-- Evolução supressão primária vs. secundária
SELECT transition_name, year, ROUND(SUM(area_ha),2) AS area_total_ha
FROM gold.fact_desmatamento g
JOIN gold.dim_transicao m
ON g.transicao_id = m.transicao_id
GROUP BY year, transition_name
ORDER BY year, transition_name

transition_name year area_total_ha Supressão Veg. Primária 1987 5647033.74 Supressão Veg. Secundária 1987 0.0 Supressão Veg. Primária 1988 5580900.85 Supressão Veg. Secundária 1988 0.0 Supressão Veg. Primária 1989 5124286.32 Supressão Veg. Secundária 1989 33998.7 Supressão Veg. Primária 1990 4753595.85 Supressão Veg. Secundária 1990 94525.17 Supressão Veg. Primária 1991 4691199.96 Supressão Veg. Secundária 1991 193044.21 Supressão Veg. Primária 1992 5380111.49 Supressão Veg. Secundária 1992 380381.54 Supressão Veg. Primária 1993 4765603.86 Supressão Veg. Secundária 1993 456889.3 Supressão Veg. Primária 1994 4922418.14 Supressão Veg. Secundária 1994 537091.41 Supressão Veg. Primária 1995 5517778.62 Supressão Veg. Secundária 1995 706532.93 Supressão Veg. Primária 1996 4918441.68 Supressão Veg. Secundária 1996 693045.95 Supressão Veg. Primária 1997 4680413.3 Supressão Veg. Secundária 1997 716347.13 Supressão Veg. Primária 1998 4701191.91 Supressão Veg. Secundária 1998 784462.57 Supressão Veg. Primária 1999 4594195.03 Supressão Veg. Secundária 1999 821951.53 Supressão Veg. Primária 2000 4159871.02 Supressão Veg. Secundária 2000 819257.07 Supressão Veg. Primária 2001 4098664.21 Supressão Veg. Secundária 2001 809644.07 Supressão Veg. Primária 2002 4285200.06 Supressão Veg. Secundária 2002 811611.41 Supressão Veg. Primária 2003 5422519.9 Supressão Veg. Secundária 2003 959990.23 Supressão Veg. Primária 2004 5288431.67 Supressão Veg. Secundária 2004 978666.11 Supressão Veg. Primária 2005 4311530.49 Supressão Veg. Secundária 2005 829219.51 Supressão Veg. Primária 2006 3533587.51 Supressão Veg. Secundária 2006 820971.59 Supressão Veg. Primária 2007 2945501.53 Supressão Veg. Secundária 2007 845981.39 Supressão Veg. Primária 2008 2981830.92 Supressão Veg. Secundária 2008 918946.27 Supressão Veg. Primária 2009 2276241.58 Supressão Veg. Secundária 2009 827542.31 Supressão Veg. Primária 2010 2133038.29 Supressão Veg. Secundária 2010 861000.28 Supressão Veg. Primária 2011 2161794.9 Supressão Veg. Secundária 2011 923871.56 Supressão Veg. Primária 2012 2440620.8 Supressão Veg. Secundária 2012 1068956.61 Supressão Veg. Primária 2013 2732853.93 Supressão Veg. Secundária 2013 1172093.31 Supressão Veg. Primária 2014 2369552.93 Supressão Veg. Secundária 2014 1079931.43 Supressão Veg. Primária 2015 2527822.59 Supressão Veg. Secundária 2015 1112686.06 Supressão Veg. Primária 2016 2338580.76 Supressão Veg. Secundária 2016 1055473.17 Supressão Veg. Primária 2017 2141118.05 Supressão Veg. Secundária 2017 999481.23 Supressão Veg. Primária 2018 2191196.33 Supressão Veg. Secundária 2018 1020585.92 Supressão Veg. Primária 2019 3050405.57 Supressão Veg. Secundária 2019 1183767.42 Supressão Veg. Primária 2020 3109857.4 Supressão Veg. Secundária 2020 1136998.39 Supressão Veg. Primária 2021 3291965.66 Supressão Veg. Secundária 2021 1273868.71 Supressão Veg. Primária 2022 3533411.24 Supressão Veg. Secundária 2022 1478729.31 Supressão Veg. Primária 2023 3374733.94 Supressão Veg. Secundária 2023 1660681.43 Supressão Veg. Primária 2024 2890982.24 Supressão Veg. Secundária 2024 1291838.79

## Conclusões

- Como era de se esperar, o bioma com maior volume de desmatamento acumulado é a Amazônia. No entanto, em 2023 e 2024, seu desmatamento caiu drasticamente e o Cerrado a ultrapassou como bioma mais desmatado do ano.
- O desmatamento da Caatinga tem crescido bastante desde 2020, também superando a Amazônia em 2024.
- A vegetação mais desmatada é a Formação Florestal.
- O estado do Mato Grosso apresenta o maior volume de desmatamento, tanto em volume absoluto de vegetação nativa (primária) desmatada, quanto no total acumulado. É também o estado com a maior porcentagem de sua vegetação nativa desmatada (cerca de 45%), seguido de Rondônia, Tocantins e Maranhão.
- A supressão de vegetação primária representa a maior parcela do desmatamento anual, mas a secundária tem crescido continuamente nos últimos anos.

## Próximos passos

- Incorporar dados de políticas públicas e eventos nacionais (ex: criação de áreas protegidas, campanhas de fiscalização, mandatos presidenciais, etc.) para avaliar correlações com o desmatamento.
- Cruzar com dados da COMEX (comércio exterior) para investigar possível vínculo entre desmatamento e exportação de commodities
- Estender a análise com outras bases de dados do MapBiomas, como Cicatrizes de Fogo, Mineração e Agricultura.